pkg/processor/eventprocessor/pool.go

```diff
@@ -86,21 +90,35 @@ func (sa *syncPoolAllocator) Allocate(timeout time.Duration) (EventProcessor, er
 	}
 }
 
-func (sa *syncPoolAllocator) Stop() {
+func (sa *syncPoolAllocator) Stop() error {
 	// Stop the old objects that are being cleaned up
-	for _, object := range sa.objects {
-		if err := object.Stop(); err != nil {
-			sa.logger.WarnWith("Failed to stop object",
-				"error", err)
-		}
+	if err := sa.SignalTermination(); err != nil {
+		sa.logger.DebugWith("Failed to stop objects in allocator",
+			"error", err.Error())
 	}
 
+	// close channel
 	if sa.objectsChan != nil {
 		close(sa.objectsChan)
 	}
+	// clean up objects
+	clear(sa.objects)
+	return nil
 }
 
 func (sa *syncPoolAllocator) Release(object EventProcessor) {
+	if sa.IsTerminated() {
+		sa.logger.DebugWith("Allocator is terminated, not releasing object",
+			"object", object.GetIndex())
+		return
+	}
```

Comment:
Why do you skip it in this case?
Because it can still handle events?

Reply:
@TomerShor no, it can't. If allocator is terminated, chan would usually be closed already.